### PR TITLE
webapi and vnmgr services run with own user

### PR DESCRIPTION
**Pull request by Mopster**
The webapi and vnmgr have gotten their own user. The rpm creates the users and adds writing permissions to the log files. The upstart scripts start the service with the webapi and vnmgr user.

### CODE REVIEW

File: deployment/packagebuild/packages.d/vnet/openvnet-webapi/recipe.rb [hunk 1]
```diff
@@ -15,13 +15,18 @@ class OpenvnetWebapi < FPM::Cookery::Recipe
     /etc/openvnet/webapi.conf
   )
 
+  post_install 'post-install'
+
   def build
   end
 
   def install
+    puts "*" * 50
```
Comment (hansode) [inline deployment/packagebuild/packages.d/vnet/openvnet-webapi/recipe.rb]: [reek] [DuplicateMethodCall] calls "*" * 50 2 times

File: deployment/packagebuild/packages.d/vnet/openvnet-webapi/recipe.rb [hunk 2]
```diff
@@ -15,13 +15,18 @@ class OpenvnetWebapi < FPM::Cookery::Recipe
...
+    puts "Install OpenVNet webapi"
     opt('axsh/openvnet/vnet').install Dir["vnet/rack"]
     etc('/init').install Dir['deployment/conf_files/etc/init/vnet-webapi.conf']
     etc('/default').install Dir['deployment/conf_files/etc/default/vnet-webapi']
     etc('/openvnet').install Dir['deployment/conf_files/etc/openvnet/webapi.conf']
+    puts "*" * 50
```
Comment (hansode) [inline deployment/packagebuild/packages.d/vnet/openvnet-webapi/recipe.rb]: [reek] [DuplicateMethodCall] calls "*" * 50 2 times

File: deployment/packagebuild/packages.d/vnet/openvnet-webapi/recipe.rb (hunk 1 above)
Comment (hansode) [inline deployment/packagebuild/packages.d/vnet/openvnet-webapi/recipe.rb]: [reek] [DuplicateMethodCall] calls puts("*" * 50) 2 times

File: deployment/packagebuild/packages.d/vnet/openvnet-webapi/recipe.rb (hunk 2 above)
Comment (hansode) [inline deployment/packagebuild/packages.d/vnet/openvnet-webapi/recipe.rb]: [reek] [DuplicateMethodCall] calls puts("*" * 50) 2 times

**Comment by unakatsuo** [comment]
I think it does not need to have individual users per process. it can share one common system account such like "openvnet".

**Comment by Mopster** [comment]
CI integration test results

18:15:22 --------------------------------------------------
18:15:22 simple: success
18:15:22 router_v2v: success
18:15:22 router_p2v: failure
18:15:22 event: success
18:15:22 service: success
18:15:22 secg: success
18:15:22 secg_reference: success
18:15:22 edge: failure
18:15:22 --------------------------------------------------

File: deployment/packagebuild/packages.d/vnet/openvnet-vnmgr/post-install [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+set -e
+
+user="vnet-vnmgr"
+logfile="/var/log/openvnet/vnmgr.log"
+
+if ! id "$user" > /dev/null 2>&1 ; then
+    adduser --system --no-create-home --shell /bin/false "$user"
+fi
+
+touch "$logfile"
```
Comment (unakatsuo) [inline deployment/packagebuild/packages.d/vnet/openvnet-vnmgr/post-install]: it may fail if the folder "/var/log/openvnet/" was missing.

Comment (Mopster) [inline deployment/packagebuild/packages.d/vnet/openvnet-vnmgr/post-install]: the openvnet-common recipe creates the folder : https://github.com/axsh/openvnet/blob/master/deployment/packagebuild/packages.d/vnet/openvnet-common/recipe.rb#L44

**Comment by Metallion** [comment]
@unakatsuo I disagree with having a single user for all processes. Every process has its own specific responsibilities and therefore each process has different user permission requirements.

For example there is no reason for the Web API to be able to access Open vSwitch. Giving it rights it doesn't need qualifies as a security issue.

Also having different users helps us enforce the separation of responsibilities between processes. They make it impossible for a process to perform a task it wasn't intended to do.

File: deployment/packagebuild/packages.d/vnet/openvnet-vnmgr/post-install (hunk 3 above)
Comment (unakatsuo) [inline deployment/packagebuild/packages.d/vnet/openvnet-vnmgr/post-install]: ok. then each upstart conf can remove the lines:

```
    [ -d "${LOG_DIRECTORY}" ] || {
      mkdir ${LOG_DIRECTORY}
    }
```

File: deployment/conf_files/etc/init/vnet-vnmgr.conf [hunk 4]
```diff
@@ -8,6 +8,10 @@ respawn limit 5 60
 env NAME=vnmgr
 
 script
+    ## log
+    #exec 2>>/var/log/openvnet/vnmgr-upstart.log
+    #set -x
```
Comment (Metallion) [inline deployment/conf_files/etc/init/vnet-vnmgr.conf]: Commented out code shouldn't be committed. Please remove it.

File: deployment/conf_files/etc/init/vnet-vnmgr.conf [hunk 5]
```diff
@@ -21,6 +25,5 @@ script
     }
     cd ${VNET_ROOT}/vnet/
 
-    exec bundle exec ./bin/${NAME} \
-        >> ${LOG_DIRECTORY}/${NAME}.log 2>&1
+    su ${VNET_USER} -s /bin/sh -c "exec bundle exec ./bin/${NAME}" >> ${LOG_DIRECTORY}/${NAME}.log 2>&1
```
Comment (Metallion) [inline deployment/conf_files/etc/init/vnet-vnmgr.conf]: I think we should use `/bin/bash` as our shell.

File: deployment/packagebuild/packages.d/vnet/openvnet-vnmgr/post-install (hunk 3 above)
Comment (Metallion) [inline deployment/packagebuild/packages.d/vnet/openvnet-vnmgr/post-install]: +1

If the openvnet-common RPM creates the folder, then the upstart jobs don't need to check for it. It's kind of unrelated to the problem at hand though. Can either be fixed here or in a next pull request.

File: deployment/packagebuild/packages.d/vnet/openvnet-vnmgr/recipe.rb [hunk 6]
```diff
@@ -18,11 +18,14 @@ class OpenvnetVnmgr < FPM::Cookery::Recipe
     /etc/openvnet/vnmgr.conf
   )
 
+  post_install 'post-install'
```
Comment (Metallion) [inline deployment/packagebuild/packages.d/vnet/openvnet-vnmgr/recipe.rb]: Can you change the file name to `post-install.sh`? That will make it more clear that it's a shell script.

File: deployment/conf_files/etc/init/vnet-vnmgr.conf (hunk 4 above)
Comment (Mopster) [inline deployment/conf_files/etc/init/vnet-vnmgr.conf]: it's a good way to keep this piece code for debugging purposes, so you don't need to spend time trying to find out how to enable logging the upstart script again it's only 3 lines

File: deployment/packagebuild/packages.d/vnet/openvnet-webapi/recipe.rb (hunk 2 above)
Comment (Metallion) [inline deployment/packagebuild/packages.d/vnet/openvnet-webapi/recipe.rb]: These reeks are fine. The calls are so simple that a separate method for them would be worse.

File: deployment/conf_files/etc/init/vnet-vnmgr.conf (hunk 4 above)
Comment (Metallion) [inline deployment/conf_files/etc/init/vnet-vnmgr.conf]: True. In that case can you change the first line to something like this? `## Uncomment to enable upstart debug logging.`

**Pull request merged**